### PR TITLE
Add fastnetip IPv4 and IPv6 containment helpers to BART lookups

### PR DESCRIPTION
**Pull request by gaissmai**
@coderabbit

## Summary by CodeRabbit

* **Bug Fixes**
  * IPv4 and IPv6 prefix matching now handles address families more reliably.
  * Zoned IPv6 address behavior is documented as undefined for containment checks and lookups.

* **Performance**
  * Prefix containment checks are optimized for faster lookups without changing standard results.

* **Tests**
  * Expanded coverage for IPv4/IPv6 boundary cases, prefix matching, and supported build configurations.
  * Added benchmarks for containment checks.

### CODE REVIEW

**Comment by gaissmai** [comment]
@coderabbit resume

**Comment by gaissmai** [comment]
@coderabbitai full review

File: internal/fastnetip/netip_unsafe.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+// It mirrors the unexported uint128 type used internally by net/netip.
+type uint128 struct {
+	hi uint64
+	lo uint64
+}
+
+// Addr mirrors the memory layout of net/netip.Addr.
+type Addr struct {
+	addr uint128
+	z    uintptr // Mirrors unique.Handle[addrDetail]
```
Comment (database64128) [inline internal/fastnetip/netip_unsafe.go]: Minor nit: Strictly speaking this should be `z unsafe.Pointer` (or any other pointer type to a value type that's no larger than `addrDetail`). The GC does not treat pointer values in `uintptr` as references, so if the original `netip.Addr.z` is no longer reachable in any other place, the unique handle becomes eligible for GC reclamation as soon as the conversion to our local `Addr` or `Prefix` type happens. This is documented in [`unsafe.Pointer`'s doc comment](https://pkg.go.dev/unsafe#Pointer).

Although in this specific use case `uintptr` is fine, as long as we don't use `z` via this `Addr` type.

**Review by database64128 — COMMENTED** [review]
Nice! This reminds me of something I did in https://github.com/database64128/shadowsocks-go/blob/main/conn/addr.go, although mine was arguably much less justifiable :P

File: internal/fastnetip/netip_unsafe.go (hunk 1 above)
Comment (gaissmai) [reply internal/fastnetip/netip_unsafe.go]: @database64128 thanks for your review!
For now, this is just a proof of concept to test with real data whether the added code complexity actually pays off in real-world performance gains.
